python/dgl/nodeflow.py

```diff
@@ -17,6 +17,96 @@
 
 NodeFlowHandle = ctypes.c_void_p
 
+class NodeFlowIndex(GraphIndex):
```

Comment:
Do we need a `NodeFlowIndex` object? During implementation, I found it unnecessary for most cases. The `block_edges` and `block_adjacency_matrix` could be put in `NodeFlow` directly. If they are internal functions that should not be seen by users. We could make them global functions and pass the necessary arguments to it.

Reply:
we need. it's mainly used by schedulers and the runtime. These components rely on graph index.

Reply:
Graph index should be accessible using `nodeflow._graph` right? My point is that all the members of this `NodeFlowIndex` is passed by the `NodeFlow` class, so why bother maintaining another class? They can be put in `NodeFlow` class and accessible from there.

Reply:
but these components don't have a reference to NodeFlow.

Reply:
Oh, you want to have `schedule_nodeflow_compute` to reuse schedule logics from `DGLGraph`? but I saw the `self` object has been passed to the function so scheduler and runtime could hold a reference to NodeFlow right?

Reply:
Another concern is duplicated logic. For example what's the difference of `layer_size` in L25 and L224?

Reply:
oh, i see. you always pass DGLGraph around.

Reply:
i see a case that NodeFlowIndex is important. In the distributed sampler, it's much easier to have NodeFlowIndex around.

Reply:
Ok? Could you explain a little bit?

Reply:
I just discussed with @aksnzhy, the distributed sampler works fine without NodeFlowIndex.